### PR TITLE
[DT-322] Migrate to the new install manager

### PR DESCRIPTION
**Pull request by G0retZ**
**What does this PR do?**

   Add vanilla implementation for the install manager to install apps with tests.

**Database changed?**

   Yes

**Where should the reviewer start?**

- [ ] Better by commits

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [DT-322](https://aptoide.atlassian.net/browse/DT-322)

**What are the relevant tickets?**

  Tickets related to this pull-request: [DT-322](https://aptoide.atlassian.net/browse/DT-322)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[DT-322]: https://aptoide.atlassian.net/browse/DT-322?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DT-322]: https://aptoide.atlassian.net/browse/DT-322?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: aptoide-installer/src/main/java/cm/aptoide/pt/installer/platform/File.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import android.text.TextUtils
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.math.BigInteger
+import java.security.MessageDigest
+import java.security.NoSuchAlgorithmException
+
+
```
Comment (pedroagostinhoaptoide) [inline aptoide-installer/src/main/java/cm/aptoide/pt/installer/platform/File.kt:12]: The file could have a better name, like 'FileExtensions' for example

File: aptoide-installer/src/main/java/cm/aptoide/pt/installer/platform/PackageInstallerResultCallback.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package cm.aptoide.pt.installer.platform
+
+interface PackageInstallerResultCallback {
```
Comment (pedroagostinhoaptoide) [inline aptoide-installer/src/main/java/cm/aptoide/pt/installer/platform/PackageInstallerResultCallback.kt:3]: Unused interface

File: aptoide-installer/src/main/java/cm/aptoide/pt/installer/platform/File.kt (hunk 1 above)
Comment (G0retZ) [reply aptoide-installer/src/main/java/cm/aptoide/pt/installer/platform/File.kt]: Fixed and force pushed

File: aptoide-installer/src/main/java/cm/aptoide/pt/installer/platform/PackageInstallerResultCallback.kt (hunk 2 above)
Comment (G0retZ) [reply aptoide-installer/src/main/java/cm/aptoide/pt/installer/platform/PackageInstallerResultCallback.kt]: Fixed and force pushed

File: install-manager/build.gradle.kts [hunk 3]
```diff
@@ -39,5 +40,5 @@ dependencies {
   implementation(LibraryDependency.KOTLIN)
 
   testRuntimeOnly(TestLibraryDependency.JUNIT_JUPITER_ENGINE)
-  testImplementation(project(ModuleDependency.UTIL))
+  testImplementation(project(ModuleDependency.TEST))
 }
```
Comment (ricardo-frango) [inline install-manager/build.gradle.kts:44]: no EOF line

**Review by ricardo-frango — CHANGES_REQUESTED** [review]

File: aptoide-installer/build.gradle.kts [hunk 4]
```diff
@@ -29,28 +30,30 @@ android {
...
+  implementation(LibraryDependency.COROUTINES)
 
   //di
   implementation(LibraryDependency.HILT)
-  implementation(LibraryDependency.HILT_NAV_COMPOSE)
   kapt(LibraryDependency.HILT_COMPILER)
 
+  testRuntimeOnly(TestLibraryDependency.JUNIT_JUPITER_ENGINE)
+  testImplementation(project(ModuleDependency.TEST))
 }
```
[the remark this answers is not in the document]
Comment (G0retZ) [reply aptoide-installer/build.gradle.kts]: Fixed

File: install-manager/build.gradle.kts (hunk 3 above)
Comment (G0retZ) [reply install-manager/build.gradle.kts]: Fixed

**Pull request merged**